src/visibility-and-privacy.md

```diff
@@ -105,17 +102,23 @@ mod crate_helper_module {
 
 // This function is "public to the root" meaning that it's available to external
 // crates linking against this one.
-pub fn public_api() {}
+pub fn public_api(rate_limit:i32) {}
+
+//This is by default private, and it isn't available to external crates.
+const MAX_RATE:i32 = 32;
 
 // Similarly to 'public_api', this module is public so external crates may look
 // inside of it.
 pub mod submodule {
-    use crate_helper_module;
-
+    use super::crate_helper_module;
+    use super::MAX_RATE;
+    use super::public_api;
+    
     pub fn my_method() {
         // Any item in the local crate may invoke the helper module's public
         // interface through a combination of the two rules above.
         crate_helper_module::crate_helper();
+        public_api(MAX_RATE);
```

Comment:
It would probably be best to have a comment indicating what this is illustrating.